Logging/tests/Unit/LoggerInterfaceTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Google\Cloud\Logging\Tests\Unit;
+
+use DateTime;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+
+abstract class LoggerInterfaceTest extends TestCase
```

Comment:
This makes me nervous... it looks like this class was copied over from the PSR tests?

All the methods you've copied over are `public`, so why wouldn't we just implemented whatever changes we need in `PsrLoggerCompatibilityTest.php` instead, since that's the only test that extends this `LoggerInterfaceTest`? 

Also, what changes are required? I can't tell since you copied everything over.

Reply:
I copied them over because `psr/log` 1.1 had a TestLogger which has been removed in 2.0 and 3.0.

This way, all `psr/log` versions are testable.

However, in the meantime, [fig/log-test](https://github.com/php-fig/log-test#versions) has been started, but support for `psr/log` 2.0/3.0 has not yet been released. 

I'm going to require that package with `^1.0|^2.0-dev` in the hope both can be used and remove the `-dev` once it's released (this should happen within the week).

Reply:
I think so, too, at least for the moment. I'll do it later tonight! 🤞🏻

Reply:
I think I'll have to wait until fig/log-test 1.1 is released - or it's too late for me, and I can't figure out how to add the requirement to use the released version or the dev-version, depending on which `psr/log` version is installed. This, at least, doesn't work for me right now:

```
"fig/log-test": "1.0.*||dev-main#1d0c40f",
```

Reply:
The PR now uses the latest release of `fig/log-test` - since the current release only supports `psr/log ^1`, `monolog/monolog` is again fixed to 2.x until `fig/log-test` receives the planned release.

I've tested it locally with the main branch, and the tests fail because the method signatures of the `PsrLogger` class are incompatible with Monolog v3, so I have some more work to do until next week.